### ISSUE TITLE
BUG in Diffusion Model During Training

### ISSUE THREAD

**Issue by MaverickRen** [nvidia-cosmos/cosmos-predict1#97]
Issue Description:

I encountered the following problems while debugging with the Video2World-7B pretrained weights and the provided cosmos_nemo_assets dataset:

1. During training, the visualization of `model_pred.x0` appears very blurry, even though I have already loaded the pretrained weights.

The visualization code is implemented in ["cosmos_predict1/diffusion/training/models/model_image.py"](https://github.com/nvidia-cosmos/cosmos-predict1/blob/5a96e9f43a579ffcd11879ed7f521c4c162da35f/cosmos_predict1/diffusion/training/models/model_image.py#L230)

`mean, std = self.sde.marginal_prob(x0, sigma)
        # Generate noisy observations
        xt = mean + batch_mul(std, epsilon)  # corrupted data
        # make prediction
        model_pred = self.denoise(xt, sigma, condition)
        # loss weights for different noise levels
        weights_per_sigma = self.get_per_sigma_loss_weights(sigma=sigma)
        # extra weight for each sample, for example, aesthetic weight, camera weight
        weights_per_sample = self.get_per_sample_weight(data_batch, x0_from_data_batch.shape[0])
        # extra loss mask for each sample, for example, human faces, hands
        loss_mask_per_sample = self.get_per_sample_loss_mask(data_batch, x0_from_data_batch.shape, x0.shape)
        pred_mse = (x0 - model_pred.x0) ** 2 * loss_mask_per_sample
        edm_loss = batch_mul(pred_mse, weights_per_sigma * weights_per_sample)

        decoded_x0 = self.vae.decode(model_pred.x0_pred_replaced)
        decoded_x0 = (((decoded_x0[0] * 0.5) + 0.5).clamp(0, 1) * 255).float().cpu().to(torch.uint8).permute(1, 2, 3, 0).numpy()
        import imageio
        imageio.mimsave(f"/opt/tiger/test_input_dit.mp4", decoded_x0, fps=30 )
`
The visualization is:
https://github.com/user-attachments/assets/d6be2a42-14bb-497a-9332-d41a1fdb2d90

2. The visualization works fine in inference mode.

The visualization is:

https://github.com/user-attachments/assets/9d554726-f023-441f-a040-2f29d9a70edd

4. Why are the training configuration files different from the testing configuration files? For example, `extra_per_block_abs_pos_emb` is set to `False` in the training configuration.

**Comment by MaverickRen** [comment]
I set the 'extra_per_block_abs_pos_emb' to 'True' when Training and find the visualization works fine. So why the training configuration files different from the testing configuration files?

This visualization  is: 

https://github.com/user-attachments/assets/453713bc-5443-45f8-949d-ee3ac3ae5ca5

**Comment by NevermoreCY** [comment]
Hi, this is issue is caused by **extra_per_block_abs_pos_emb** config, it should be the same for training and inference. It was fixed by previous commit. I tested with the latest repo with the visualization code you provide , the video generated during training looks good. 

https://github.com/user-attachments/assets/68824dc1-d4a3-4491-9eeb-845ef2992509
